driver-core/src/test/java/com/datastax/driver/core/HostConnectionPoolTest.java

```diff
@@ -1197,16 +1198,11 @@ public void should_not_mark_host_down_if_some_connections_fail_on_init() throws
 
       Uninterruptibles.sleepUninterruptibly(reconnectInterval, TimeUnit.MILLISECONDS);
 
-      // Should open up to core connections, however it will only spawn up to 1 connection
-      // per request, so we need to make enough requests to make up the deficit.  Additionally
-      // we need to wait for connections to be established between requests for the pool
-      // to spawn new connections (since it only allows one simultaneous creation).
-      for (int i = 5; i <= 8; i++) {
-        allRequests.add(MockRequest.send(pool));
-        verify(factory, timeout(readTimeout)).open(any(HostConnectionPool.class));
-        reset(factory);
-        assertPoolSize(pool, i);
-      }
+      // Reconnection mechanism should fill missing connections by now.
+      verify(factory, timeout(readTimeout).times(4))
+          .open(any(HostConnectionPool.class), anyInt(), anyInt());
+      reset(factory);
+      assertPoolSize(pool, 8);
```

Comment:
I think that the old code can be made working - by making the change in my previous comment (related to `ConnectionResult.FAILED` in one case).

Reply:
Isn't the new behavior nicer?